server/tasks/process_file.py

```diff
@@ -30,32 +34,82 @@ def process_file(serialized_agent: bytes):
-def process_file_streaming(self, serialized_agent: bytes):
-    # Load the agent
-    agent = pickle.loads(serialized_agent)
+@app.task(
+    name="streaming_parent_task", track_started=True, bind=True, serializer="pickle"
+)
+def streaming_parent_task(
+    self, agent: Agent, result_handler: ResultHandler, batch_size: int = 2
+):
+    # Parent job ID is used for input/output topic names
+    parent_job_id = self.request.id
+
+    task = process_file_streaming
+    logger.info(f"Submitting task {task.name} with agent {agent}")
+    input_result = task.delay(agent=agent, parent_job_id=parent_job_id)
+    input_job_id = input_result.id
+    logger.info(f"Task {task.name} submitted with job_id {input_job_id}")
+
+    task = process_streaming_output
+    logger.info(f"Submitting task {task.name}")
+    output_result = task.delay(
+        input_job_id=input_job_id,
+        parent_job_id=parent_job_id,
+        result_handler=result_handler,
+        batch_size=batch_size,
+    )
+    output_job_id = output_result.id
+    logger.info(f"Task {task.name} submitted with job_id {output_job_id}")
+
+    # Store input and output job IDs in parent task metadata
+    # Need to pass state as well otherwise its overwritten to None
+    self.update_state(
+        state=states.STARTED,
+        meta={"input_job_id": input_job_id, "output_job_id": output_job_id},
+    )
+
+    input_job = process_file_streaming.AsyncResult(input_job_id)
+    output_job = process_streaming_output.AsyncResult(output_job_id)
+
+    terminal_statuses = ["SUCCESS", "FAILURE", "REVOKED"]
+
+    while (
+        input_job.status not in terminal_statuses
+        or output_job.status not in terminal_statuses
+    ):
+        time.sleep(1)
 
-    # Get own job ID to set Consumer topic accordingly
-    job_id = self.request.id
-    agent.environment.kafka_input_topic = get_input_topic(job_id)
-    agent.environment.kafka_output_topic = get_output_topic(job_id)
+    logger.info("Both input and output jobs complete")
+
+    # Update parent task status to SUCCESS and pass metadata again
+    # otherwise its overwritten to None
+    self.update_state(
+        state=states.SUCCESS,
+        meta={"input_job_id": input_job_id, "output_job_id": output_job_id},
+    )
+    raise Ignore()
+
+
+@app.task(
+    name="process_file_streaming", track_started=True, bind=True, serializer="pickle"
+)
+def process_file_streaming(self, agent: Agent, parent_job_id: str):
+    # Set input and output topics using parent job ID
+    agent.environment.kafka_input_topic = get_input_topic(parent_job_id)
+    agent.environment.kafka_output_topic = get_output_topic(parent_job_id)
 
     # Run the agent
     asyncio.run(agent.arun())
 
 
 async def async_process_streaming_output(
-    input_job_id: str, result_handler: str, batch_size: int
+    input_job_id: str,
+    parent_job_id: str,
+    result_handler: ResultHandler,
+    batch_size: int,
```

Comment:
Would it make sense to provide a default `batch_size`?

Reply:
yes and it should be coming from `process_streaming_output` which was there, will add back